app/lib/handover/handover_case.rb

```diff
@@ -1,10 +1,15 @@
 class Handover::HandoverCase
   # we pass calculated handover date explicitly instead of finding it ourselves because: (1) its not our job to find it,
   # and (2) the CategorisedHandoverCases factories will have found them already when it initialises this class
-  def initialize(calculated_handover_date, allocated_offender)
-    raise ArgumentError unless calculated_handover_date.is_a?(CalculatedHandoverDate)
+  def initialize(allocated_offender, calculated_handover_date = nil)
     raise ArgumentError unless allocated_offender.is_a?(AllocatedOffender)
 
+    if calculated_handover_date
+      raise ArgumentError unless calculated_handover_date.is_a?(CalculatedHandoverDate)
+    else
+      calculated_handover_date = CalculatedHandoverDate.find_by!(nomis_offender_id: allocated_offender.offender_no)
```

Comment:
I read the descriptions in lines 34-36 as it being possible for a CalculatedHandoverDate to not exist for some offenders (due to be unable to calculate it).
With this being the case, what are the impacts of `CalculatedHandoverDate.find_by!` returning `nil` here?

Reply:
I think I'll undo that commit because this variation of the constructor is never used.... one sec.